src/translib/transformer/xfmr_system.go

```diff
@@ -0,0 +1,302 @@
+package transformer
+
+import (
+    "encoding/json"
+    "translib/ocbinds"
+    "translib/tlerr"
+    "time"
+    "io/ioutil"
+    "syscall"
+    "strconv"
+    "os"
+    log "github.com/golang/glog"
+    ygot "github.com/openconfig/ygot/ygot"
+)
+
+func init () {
+    XlateFuncBind("DbToYang_sys_state_xfmr", DbToYang_sys_state_xfmr)
+    XlateFuncBind("DbToYang_sys_memory_xfmr", DbToYang_sys_memory_xfmr)
+    XlateFuncBind("DbToYang_sys_cpus_xfmr", DbToYang_sys_cpus_xfmr)
+    XlateFuncBind("DbToYang_sys_procs_xfmr", DbToYang_sys_procs_xfmr)
+}
+
+type JSONSystem  struct {
+    Hostname       string  `json:"hostname"`
+    Total          uint64  `json:"total"`
+    Used           uint64  `json:"used"`
+    Free           uint64  `json:"free"`
+
+    Cpus  []Cpu            `json:"cpus"`
+    Procs map[string]Proc  `json:"procs"`
+
+}
+
+type Cpu struct {
+    User     int64   `json:"user"`
+    System   int64   `json:"system"`
+    Idle     int64   `json:"idle"`
+}
+
+type Proc struct {
+    Cmd        string     `json:"cmd"`
+    Start      uint64     `json:"start"`
+    User       uint64     `json:"user"`
+    System     uint64     `json:"system"`
+    Mem        uint64     `json:"mem"`
+    Cputil   float32    `json:"cputil"`
+    Memutil   float32    `json:"memutil"`
+}
+
+type CpuState struct {
+    user uint8
+    system uint8
+    idle   uint8
+}
+
+type ProcessState struct {
+    Args [] string
+    CpuUsageSystem uint64
+    CpuUsageUser   uint64
+    CpuUtilization uint8
+    MemoryUsage    uint64
+    MemoryUtilization uint8
+    Name              string
+    Pid               uint64
+    StartTime         uint64
+    Uptime            uint64
+}
+
+func getAppRootObject(inParams XfmrParams) (*ocbinds.OpenconfigSystem_System) {                                                                  
+    deviceObj := (*inParams.ygRoot).(*ocbinds.Device)
+    return deviceObj.System                                                                                                                 
+}            
+
+func getSystemInfoFromFile () (JSONSystem, error) {
+    log.Infof("getSystemInfoFromFile Enter")
+
+    var jsonsystem JSONSystem 
+    jsonFile, err := os.Open("/mnt/platform/system")
+    if err != nil {
+        log.Infof("system json open failed")
+        errStr := "Information not available or Platform support not added"
+        terr := tlerr.NotFoundError{Format: errStr}
+        return jsonsystem, terr
+    }
+    syscall.Flock(int(jsonFile.Fd()),syscall.LOCK_EX)
+    log.Infof("syscall.Flock done")
+
+    defer jsonFile.Close()
+    defer log.Infof("jsonFile.Close called")
+    defer syscall.Flock(int(jsonFile.Fd()), syscall.LOCK_UN);
+    defer log.Infof("syscall.Flock unlock  called")
+
+    byteValue, _ := ioutil.ReadAll(jsonFile)
+    json.Unmarshal(byteValue, &jsonsystem)
```

Comment:
Can you please check a parse error?
err := json.Unmarshal(byteValue, &jsonsystem)
if err != nil {
    . . .
}